felix/calc/policy_lookup_cache.go

```diff
@@ -567,61 +567,6 @@ func (r *RuleID) GetFlowLogPolicyName() string {
 	return r.fpName
 }
 
-// deconstructPolicyName deconstructs the v1 policy name that is constructed by the SyncerUpdateProcessors in
-// libcalico-go and extracts the v3 fields: namespace, tier, name.
-//
-// The v1 policy name is of the format:
-// -  <namespace>/<tier>.<name> for a namespaced NetworkPolicies
-// -  <tier>.<name> for GlobalNetworkPolicies.
-// -  <namespace>/knp.default.<name> for a k8s NetworkPolicies
-// and for the staged counterparts, respectively:
-// -  <namespace>/staged:<tier>.<name>
-// -  staged:<tier>.<name>
-// -  <namespace>/staged:knp.default.<name>
-//
-// The namespace is returned blank for GlobalNetworkPolicies.
-// For k8s network policies, the tier is always "default" and the name will be returned including the
-// knp.default prefix.
-//
-// Staged policies will have the simplified name prefixed with "staged:", eg:
-// - <namespace>/staged:<tier>.<name>      => Name=staged:<name>, Namespace=<namespace>, Tier=<tier>
-// - staged:<tier>.<name>                  => Name=staged:<name>, Namespace=<namespace>, Tier=<tier>
-// - <namespace>/staged:knp.default.<name> => Name=staged:knp.default.<name>, Namespace=<name>, Tier=default
-func deconstructPolicyName(name string) (string, string, string, error) {
-	var namespace string
```

Comment:
Moving it to libcalico-go to use in other places as well. This is not done in this PR, and I think it make sense to move now. In follow up PRs, I'm going to use this function.